internal/header/feature_set.go

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package header
+
+// FeatureSet represents a collection of feature tags and their values.
+type FeatureSet struct {
+	features map[FeatureTag][]FeatureTagValue
```

Comment:
should consider changing to `type FeatureSet map[FeatureTag][]FeatureTagValue`.

Reply:
recommendation still stands, but should also know that there is logic to the key equality that should be considered. perhaps could introduce “canonicalize” concept like with HTTP header keys.